### PR TITLE
Catch up with SFML 3 API changes

### PR DESCRIPTION
**Pull request by ChrisThrasher**
Since we last updated CSFML, SFML has changed the `sf::Event` and Audio APIs.

### CODE REVIEW

File: src/SFML/ConvertEvent.h [hunk 1]
```diff
@@ -37,84 +37,122 @@
...
+    {
+        event->type         = sfEvtKeyReleased;
+        event->key.code     = static_cast<sfKeyCode>(keyReleased->code);
+        event->key.scancode = static_cast<sfScancode>(keyReleased->scancode);
+        event->key.alt      = keyReleased->alt;
+        event->key.control  = keyReleased->control;
+        event->key.shift    = keyReleased->shift;
+        event->key.system   = keyReleased->system;
+    }
+    else if (const auto* keyPressed = sfmlEvent.getIf<sf::Event::KeyReleased>())
```
Comment (kimci86) [inline src/SFML/ConvertEvent.h:61]:
```suggestion
    else if (const auto* keyPressed = sfmlEvent.getIf<sf::Event::KeyPressed>())
```

File: src/SFML/ConvertEvent.h
Comment (kimci86) [inline src/SFML/ConvertEvent.h:1]: Missing unit types: `Closed`, `FocusLost`, `FocusGained`, `MouseEntered`, `MouseLeft`.

File: src/SFML/Audio/SoundBuffer.cpp [hunk 2]
```diff
@@ -80,11 +80,19 @@ sfSoundBuffer* sfSoundBuffer_createFromStream(sfInputStream* stream)
 
 
 ////////////////////////////////////////////////////////////
-sfSoundBuffer* sfSoundBuffer_createFromSamples(const int16_t* samples, uint64_t sampleCount, unsigned int channelCount, unsigned int sampleRate)
+sfSoundBuffer* sfSoundBuffer_createFromSamples(const int16_t*    samples,
```
Comment (kimci86) [inline src/SFML/Audio/SoundBuffer.cpp:83]: Corresponding declaration was not updated.

File: src/SFML/Audio/SoundRecorder.cpp [hunk 3]
```diff
@@ -75,13 +75,6 @@ bool sfSoundRecorder_isAvailable(void)
 }
 
 
-////////////////////////////////////////////////////////////
-void sfSoundRecorder_setProcessingInterval(sfSoundRecorder* soundRecorder, sfTime interval)
```
Comment (kimci86) [inline src/SFML/Audio/SoundRecorder.cpp:79]: Corresponding declaration was not removed.

File: src/SFML/Audio/SoundStream.cpp [hunk 4]
```diff
@@ -35,9 +35,11 @@ sfSoundStream* sfSoundStream_create(sfSoundStreamGetDataCallback onGetData,
                                     sfSoundStreamSeekCallback    onSeek,
                                     unsigned int                 channelCount,
                                     unsigned int                 sampleRate,
-                                    void*                        userData)
+                                    void*                        userData,
+                                    sfSoundChannel*              channelMapData,
+                                    size_t                       channelMapSize)
```
Comment (kimci86) [inline src/SFML/Audio/SoundStream.cpp:40]: Corresponding declaration was not updated.
Also I am wondering if maybe `userData` should remain the last parameter.

File: src/SFML/Audio/SoundBuffer.cpp (hunk 2 above)
Comment (ChrisThrasher) [reply src/SFML/Audio/SoundBuffer.cpp:83]: I'm used to C++ where changing a member function without updating the declaration is a compiler error but alas these are not member functions.

File: src/SFML/ConvertEvent.h
Comment (ChrisThrasher) [reply src/SFML/ConvertEvent.h:1]: These were missing before. I wonder if that was intentional or if CSFML just never supported them.

Comment (ChrisThrasher) [reply src/SFML/ConvertEvent.h:1]: Oh some of those weren't missing before. By casting the event type from the SFML type to the CSFML event type those events with no data members were being handled. Okay I see now. I'll add them.

File: src/SFML/Audio/SoundStream.cpp (hunk 4 above)
Comment (ChrisThrasher) [reply src/SFML/Audio/SoundStream.cpp:40]: Why should `userData` remain the last parameter?

Comment (ChrisThrasher) [reply src/SFML/Audio/SoundStream.cpp:40]: Sounds good. I'll make that change.

**Review by eXpl0it3r — APPROVED** [review]

**Pull request merged**